### DISCUSSION TITLE
TO-DO

### DISCUSSION THREAD

**Discussion by aprilahijriyan** [zemfrog/zemfrog#86] [Ideas]
# This is a zemfrog to-do list.

## Needs improvement

* Project Layouts:

- [x] Rename `services` directory to `tasks`

* In [generator.py](https://github.com/zemfrog/zemfrog/blob/master/zemfrog/generator.py) there are 2 tasks, including:

- [x] Don't move a single template, it's better to just copy the data.

- [x] It is better to use `jinja.Template` instead of a `string.Template` to render the template.

* In [helper.py](https://github.com/zemfrog/zemfrog/blob/master/zemfrog/helper.py#L97) there are 2 tasks, including:

- [x] Removes the `db` parameter in all `db_*` functions, and

- [x] Create a local proxy to get the current db session?

- [ ] Documentation

## Additional features

- [x] Added error handling.

- [x] Added `CONTEXT_PROCESSORS` and `JINJA_FILTERS` configurations. See [here](https://flask.palletsprojects.com/en/1.1.x/templating/)

- [x] Added a new command to create a new boilerplate. (e.g. `extensions`, `models`,` tasks`, etc.)

- [x] Command to manage users and roles

- [x] Multiple static files

- [x] Role-based access control

Right now I just think of some of the ideas above.

Thank you :pray:

**Comment by mochidaz** [comment]
How about admin page template? Is that necessary?

**Comment by aprilahijriyan** [comment]
@mochidaz Of course bro :)
